sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -302,6 +308,16 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
         // general unresolved check below to throw a more tailored error message.
         new ResolveReferencesInAggregate(catalogManager).checkUnresolvedGroupByAll(operator)
 
+        getAllExpressions(operator).foreach(_.foreach {
+          case se: ScopedExpression if !se.resolved =>
```

Comment:
can we also fail if `ScopedExpression` is still present? I think the analyzer should have removed it if its child is resolved, but just in case if bugs.

Reply:
Added in b0510ff7bfd.